documentation/developer_hints/VaultResourceReference.md

```diff
@@ -0,0 +1,29 @@
+# The Vault ResourceRef
+
+## Preconditions
+
+Parameters that have a ResourceReference of type `vaultSecret` will be looked up from vault when all of the following things are true...
+
+* The environment variables `PIPER_vaultRoleID` and `PIPER_vaultRoleSecretID` must both be set to the Vault AppRole role ID and to the Vault AppRole secret ID. See [Vault Approle docs](https://www.vaultproject.io/docs/auth/approle)
+* `vaultAddress` ist set in the `general` section of the configuration file.
```

Comment:
If it's not yet too late I would vote for using `vaultServerUrl` here as we do for other services.

Reply:
It's never to late :)